src/main/java/appeng/util/helpers/P2PHelper.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * This file is part of Applied Energistics 2.
+ * Copyright (c) 2013 - 2015, AlgorithmX2, All rights reserved.
+ *
+ * Applied Energistics 2 is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Applied Energistics 2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Applied Energistics 2.  If not, see <http://www.gnu.org/licenses/lgpl>.
+ */
+
+package appeng.util.helpers;
+
+
+import com.google.common.base.Preconditions;
+
+import appeng.api.util.AEColor;
+
+
+public class P2PHelper
+{
+
+	public AEColor[] toColours( short frequency )
+	{
+		final AEColor[] colours = new AEColor[4];
+
+		for( int i = 0; i < 4; i++ )
+		{
+			int twoBits = ( frequency >> 4 * ( 3 - i ) ) & 0xF;
```

Comment:
I cannot recall how it was named, but it needs to be a nibble. So `0xF` is correct.